runtime/src/main/java/org/corfudb/runtime/collections/StreamManager.java

```diff
@@ -0,0 +1,225 @@
+                        final int defaultTimeoutInMilli = 50;
+                        TimeUnit.MILLISECONDS.sleep(defaultTimeoutInMilli);
+                    }
+                    ILogData logData = txnStream.nextUpTo(Address.MAX);
+                    MultiObjectSMREntry multiObjSMREntry = (MultiObjectSMREntry) logData.getPayload(runtime);
+                    long epoch = logData.getEpoch();
+                    Map<TableSchema, List<CorfuStreamEntry>> entries = new HashMap<>();
+                    // first only filter by the stream IDs of interest that are present in this logData
+                    logData.getStreams().stream().filter(tablesOfInterest::containsKey)
+                            .forEach(streamId -> entries.put(tablesOfInterest.get(streamId),
+                                    // Only extract the list of updates per stream as a list
+                                    multiObjSMREntry.getSMRUpdates(streamId).stream().map(smrEntry ->
+                                            CorfuStreamEntry.fromSMREntry(smrEntry,
+                                                    epoch,
+                                                    tablesOfInterest.get(streamId).getKeyClass(),
+                                                    tablesOfInterest.get(streamId).getPayloadClass(),
+                                                    tablesOfInterest.get(streamId).getMetadataClass())
+                                    ).collect(Collectors.toList())));
+
+                    if (!entries.isEmpty()) {
+                        CorfuStreamEntries callbackResult = new CorfuStreamEntries(entries);
+                        log.debug("{}::onNext with {} updates", listener.toString(), entries.size());
+                        long onNextStart = System.nanoTime();
+                        listener.onNext(callbackResult);
+                        long onNextEnd = System.nanoTime();
+                        if (TimeUnit.NANOSECONDS.toSeconds(onNextEnd - onNextStart) > 0) {
+                            log.debug("{}::onNext took {}s", listener.toString(),
+                                    TimeUnit.NANOSECONDS.toSeconds(onNextEnd - onNextStart));
+                        }
+                    }
+                }
+            } catch (Throwable throwable) {
+                log.warn("{}::onError: {}", listener.toString(), throwable.toString());
+                listener.onError(throwable);
+                streamManager.unsubscribe(listener);
+            }
+        }
+
+        public SubscriptionThread(@Nonnull StreamManager streamManager,
+                                  @Nonnull StreamListener listener,
+                                  @Nonnull String namespace,
+                                  @Nonnull List<TableSchema> tableSchemas,
+                                  long startAddress) {
+            super(String.format("CorfuSubscriber: {}:{}", listener.toString(), namespace));
```

Comment:
Sorry my bad, the format has to be
`super(String.format("CorfuSubscriber: %s:%s", listener, namespace));`

Reply:
done. thanks